packages/ember-application/tests/system/instance_initializers_test.js

```diff
@@ -1,410 +1,386 @@
 import { run } from 'ember-metal';
-import Application from '../../system/application';
-import ApplicationInstance from '../../system/application-instance';
+import { assign } from 'ember-utils';
+import { moduleFor, AutobootApplicationTestCase, TestResolver } from 'internal-test-helpers';
+import { Application, ApplicationInstance } from 'ember-application';
 import { jQuery } from 'ember-views';
 
-let app;
+moduleFor('Ember.Application instance initializers', class extends AutobootApplicationTestCase {
+  constructor() {
+    jQuery('#qunit-fixture').html(`
+      <div id="one">ONE</div>
+      <div id="two">TWO</div>
+    `);
+    super();
+  }
+
+  createApplication(options, MyApplication) {
+    let myOptions = assign({
+      rootElement: '#one'
+    }, options);
+    super.createApplication(myOptions, MyApplication);
+  }
+
+  createSecondApplication(options, MyApplication) {
+    let myOptions = assign({
+      rootElement: '#two',
+      Resolver: TestResolver
```

Comment:
This is a great start! Instead of importing the `TestResolver`, I suggest something more like [what happens here](https://github.com/emberjs/ember.js/blob/f4eef941cbb03e17f4e7eae1d52b0aa39de8605f/packages/ember-application/tests/system/application_test.js#L55-L66).

I provided a version of the code that I think works here, replacing everything from `createApplication` to `teardown`. Some notes:

* Uses `applicationOptions` instead of changing `createApplication`.
* Uses `applicationOptions` in `createSecondApplication` (avoids needing `TestResolver`.
* Default the value of `MyApplication` to `Application` in `createSecondApplication`.
* Setting `this.secondResolver` isn't needed as that property is not used.
* Prefer `runTask` over `run` (avoid the import).

```js
  get applicationOptions() {
    return assign(super.applicationOptions, {
      rootElement: '#one'
    });
  }

  createSecondApplication(options, MyApplication=Application) {
    let myOptions = assign(this.applicationOptions, {
      rootElement: '#two'
    }, options);
    let secondApp = this.secondApp = MyApplication.create(myOptions);
    return secondApp;
  }

  teardown() {
    super.teardown();

    if (this.secondApp) {
      this.runTask(() => this.secondApp.destroy());
    }
  }
```